subworkflows/local/circrna_discovery.nf

```diff
@@ -0,0 +1,227 @@
+    ch_versions = ch_versions.mix(FIND_CIRC_ANCHORS.out.versions)
+    ch_versions = ch_versions.mix(FIND_CIRC_FILTER.out.versions)
+
+    //
+    // CIRIQUANT WORKFLOW:
+    //
+
+    CIRIQUANT_YML( gtf, fasta, bwa_index.map{ meta, index -> return index }, hisat2_index )
+    CIRIQUANT( reads, CIRIQUANT_YML.out.yml.collect() )
+    CIRIQUANT_FILTER( CIRIQUANT.out.gtf.map{ meta, gtf -> meta.tool = "ciriquant"; return [ meta, gtf ] }, bsj_reads )
+
+    ch_versions = ch_versions.mix(CIRIQUANT.out.versions)
+
+    //
+    // DCC WORKFLOW
+    //
+
+    DCC_1ST_PASS( reads, star_index, gtf, true, '', '' )
+    DCC_SJDB( DCC_1ST_PASS.out.tab.map{ meta, tab -> return [ tab ] }.collect(), bsj_reads )
+    DCC_2ND_PASS( reads, star_index, DCC_SJDB.out.sjtab, true, '', '' )
+
+    mate1 = reads.map{ meta, reads -> return [meta, reads[0] ] }
+    DCC_MATE1_1ST_PASS( mate1, star_index, gtf, true, '', '' )
+    DCC_MATE1_SJDB( DCC_MATE1_1ST_PASS.out.tab.map{ meta, tab -> return [ tab ] }.collect(), bsj_reads )
+    DCC_MATE1_2ND_PASS( mate1, star_index, DCC_MATE1_SJDB.out.sjtab, true, '', '' )
+
+    mate2 = reads.map{ meta, reads -> return [ meta, reads[1] ] }
+    DCC_MATE2_1ST_PASS( mate2, star_index, gtf, true, '', '' )
+    DCC_MATE2_SJDB( DCC_MATE2_1ST_PASS.out.tab.map{ meta, tab -> return [ tab ] }.collect(), bsj_reads )
+    DCC_MATE2_2ND_PASS( mate2, star_index, DCC_MATE2_SJDB.out.sjtab, true, '', '' )
+
+    dcc_stage = DCC_2ND_PASS.out.junction.join( DCC_MATE1_2ND_PASS.out.junction, remainder: true ).join( DCC_MATE2_2ND_PASS.out.junction, remainder: true )
+    dcc = dcc_stage.map{ it -> def meta = it[0]; if( meta.single_end ){ return [ it[0], it[1], [], [] ] } else { return it } }.view()
+    DCC( dcc, fasta, gtf )
+    DCC_FILTER( DCC.out.txt.map{ meta, txt -> meta.tool = "dcc"; return [ meta, txt ] }, bsj_reads )
+
+    ch_versions = ch_versions.mix(DCC_MATE1_1ST_PASS.out.versions)
+    ch_versions = ch_versions.mix(DCC.out.versions)
+
+    //
+    // MAPSPLICE WORKFLOW:
+    //
+
+    MAPSPLICE_REFERENCE( gtf )
+    MAPSPLICE_ALIGN( reads, bowtie_index.collect(), chromosomes, gtf )
+    MAPSPLICE_PARSE( MAPSPLICE_ALIGN.out.raw_fusions )
+    MAPSPLICE_ANNOTATE( MAPSPLICE_PARSE.out.junction, fasta, MAPSPLICE_REFERENCE.out.txt )
+    mapsplice_filter = MAPSPLICE_ANNOTATE.out.txt.map{ meta, txt -> meta.tool = "mapsplice"; return [ meta, txt ] }
+    MAPSPLICE_FILTER( mapsplice_filter, bsj_reads )
+
+    ch_versions = ch_versions.mix(MAPSPLICE_REFERENCE.out.versions)
+    ch_versions = ch_versions.mix(MAPSPLICE_ALIGN.out.versions)
+    ch_versions = ch_versions.mix(MAPSPLICE_PARSE.out.versions)
+    ch_versions = ch_versions.mix(MAPSPLICE_ANNOTATE.out.versions)
+
+    //
+    // ANNOTATION WORKFLOW:
+    //
+
+    circrna_filtered = CIRCEXPLORER2_FILTER.out.results.mix(SEGEMEHL_FILTER.out.results, CIRCRNA_FINDER_FILTER.out.results, FIND_CIRC_FILTER.out.results, CIRIQUANT_FILTER.out.results, DCC_FILTER.out.results, MAPSPLICE_FILTER.out.results )
```

Comment:
This is a particularly long line, maybe split each element on one line (for readability, even though I know your style is one operation per line)